source/_cookbook/dim_and_brighten_lights.markdown

```diff
@@ -0,0 +1,164 @@
+      - service: script.turn_on
+        data_template:
+          entity_id: script.light_dim
+
+  - alias: 'Stop the dim just there'
+    initial_state: 'on'
+    trigger:
+      - platform: event
+        event_type: zwave.scene_activated
+        event_data:
+          scene_id: 16
+          entity_id: zwave.YOURCONTROLLER
+    action:
+      - service: script.turn_off
+        data_template:
+          entity_id: script.light_dim
+      - service: script.turn_off
+        data_template:
+          entity_id: script.light_dim_pause
+```
+
+There are 2 variables that control the speed of the change for the scripts below. The first is the `step` - smaller steps create a smoother transition. The second is the delay, smaller delays will create a faster transition.
+
+To allow flexibility, we'll use an [input_slider](/components/input_slider/) for the step (at the time of writing this, it's not possible to template the delay when the delay uses milliseconds). We're also using a slider to set the minimum and maximum brightness, so that it's easy to tune that (or manage it through an automation).
+
+```yaml
+input_slider:
+  light_step:
+    name: 'Step the lights this much'
+    initial: 3
+    min: 1
+    max: 64
+    step: 1
+
+  light_minimum:
+    name: 'No dimmer than this'
+    initial: 5
+    min: 1
+    max: 255
+    step: 1
+    
+  light_maximum:
+    name: 'No brighter than this'
+    initial: 255
+    min: 5
+    max: 255
+    step: 1
+```
+
+Now the scripts. There are 2 pairs of scripts. The first steps the light brighter, up to the maximum, and the second provides the delay between steps. These call each other until both are stopped. The second pair do the same for dimming.
+
+```yaml
+# Replace YOURLIGHT with the actual light entity
+script:
+    light_bright:
+      sequence:
+        - service: light.turn_on
+          data_template:
+            entity_id: light.YOURLIGHT
+            brightness: >
+              {% set current = states.light.YOURLIGHT.attributes.brightness | int %}
+              {% set step = states.input_slider.light_step.state | int %}
+              {% set next = current + step %}
+              {% if next > states.input_slider.light_maximum.state | int %}
+                {% set next = states.input_slider.light_maximum.state | int %}
+              {% endif %}
+              {{ next }}
+
+        - service: script.turn_on
+          data:
+            entity_id: script.light_bright_pause
+        
+    light_bright_pause:
+      sequence:
+        - delay:
+            milliseconds: 5
+        - service: script.turn_on
+          data:
+            entity_id: script.light_bright
+
+    light_dim:
+      sequence:
+        - service: light.turn_on
+          data_template:
+            entity_id: light.YOURLIGHT
+            brightness: >
+              {% set current = states.light.YOURLIGHT.attributes.brightness | int %}
+              {% set step = states.input_slider.light_step.state |int %}
```

Comment:
For consistency, please either remove the spaces around `|` characters (my personal preference), or add a single space after this one.